docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,121 @@
+# Guidelines for Writing Ballerina By Examples (BBEs)
+
+- [General best practices](#general-best-practices)
+- [Practices related to the code comments](#practices-related-to-the-code-comments)
+- [Practices related to the folder structure](#practices-related-to-the-folder-structure)
+
+## General best practices
+
+1. Format the Ballerina source using an IDE Plugin.
+
+2. Always, think about the length of the example. A good example should be short.
+
+3. Make the examples self-contained (i.e., avoid depending on concepts other than the one being illustrated) as much as possible.
+
+4. As a practice, use `ballerina/io` methods in main examples and `ballerina/log` in services examples. Do not use both `io:println` and `log:printInfo` in the same example.
+
+5. It is recommended to limit the maximum character count per line to 80 in .bal files. Otherwise, the lines get automatically wrapped in the [Ballerina website](https://ballerina.io/) reducing the readability as shown below.
```

Comment:
```suggestion
5. It is recommended to limit the maximum character count per line to 80 in `.bal` files. Otherwise, the lines get automatically wrapped in the [Ballerina website](https://ballerina.io/) reducing the readability as shown below.
```